### PR TITLE
Auto claim unstaking

### PR DESCRIPTION
**Pull request by AurevoirXavier**
Auto claim unstaking if it is expired while migrating.

### CODE REVIEW

File: pallet/account-migration/src/lib.rs [hunk 1]
```diff
@@ -457,14 +457,41 @@ pub mod pallet {
...
 				)?;
 
-				let sum = l.staked_kton + l.unstaking_kton.iter().map(|(k, _)| k).sum::<Balance>();
+				let sum = l.staked_kton
+					+ l.unstaking_kton
+						.iter()
+						.filter_map(
+							|(amount, expired_time)| {
+								if expired_time <= &now {
+									None
```
Comment (hackfisher) [inline pallet/account-migration/src/lib.rs:487]: Does the ledger here get updated?

Comment (AurevoirXavier) [reply pallet/account-migration/src/lib.rs:487]: I just wanted to update this. Wait a min.

Comment (hackfisher) [reply pallet/account-migration/src/lib.rs:487]: Another thing I would like to check is that, for the migrations before this upgrate, the unstakings that moved to staking pot and stayed in ledger, can they claim their unstaking?

Do we need to fix for them?

Comment (AurevoirXavier) [reply pallet/account-migration/src/lib.rs]: Yes. They can claim them immediately. @xiaoch05 did this yesterday.

**Comment by AurevoirXavier** [comment]
@boundless-forest please also help review this.

File: pallet/account-migration/src/lib.rs [hunk 2]
```diff
@@ -456,7 +456,13 @@ pub mod pallet {
 					);
 				}
 
+				let now = <frame_system::Pallet<T>>::block_number();
+
+				l.unstaking_ring.retain(|(_, t)| t > &now);
```
Comment (hackfisher) [inline pallet/account-migration/src/lib.rs:461]: wait, where do the removed unstaking_ring go? not in free right?

Comment (hackfisher) [reply pallet/account-migration/src/lib.rs:461]: OK, so free includes staked and unstaking, deposits.

Comment (boundless-forest) [reply pallet/account-migration/src/lib.rs:461]: The unstaking part belong to the user now. It'll transfer to the `staking_pot` account later.

**Review by boundless-forest — APPROVED** [review]

**Pull request merged**